src/test/java/com/networknt/schema/utils/DefaultsTest.java

```diff
@@ -31,4 +35,19 @@ void testGetDefaultNodeWhenDefaultNotFound() throws Exception {
         // Assert that the result is null, as there's no "default" node in the schema
         assertNull(result, "Default node should be null when 'default' node is not found in the schema");
     }
+
+    @Test
+    void testGetDefaultNodeWhenDefaultInRef() throws Exception {
```

Comment:
Don't create external resources for the test. Just use strings inline as it makes the test hard to read and verify. This test isn't even valid. You have a unused variable `JsonNode mainSchemaNode`.